doc/specs/stdlib_string_type.md

```diff
@@ -0,0 +1,1607 @@
+pos = scan(string, "ao")
+! pos == 2
+
+pos = scan(string, "ao", .true.)
+! pos == 6
+
+pos = scan(string, "c++")
+! pos == 0
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Verify function
+
+Scan a string for the absence of a set of characters. Verifies that all
+the characters in string belong to the set of characters in set.
+
+#### Description
+
+If *back* is either absent or *false*, this function returns the position
+of the leftmost character of *string* that is not in *set*. If *back* is *true*,
+the rightmost position is returned. If all characters of *string* are found
+in *set*, the result is zero.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):verify(interface)]] (string, set[, back])`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+- `string`: Either scalar character value or string type. This argument is `intent(in)`.
+- `set`: Either scalar character value or string type. This argument is `intent(in)`.
+- `back`: Either absent or a scalar logical value. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a default integer scalar value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+integer :: pos
+
+string = "fortran"
+pos = verify(string, "ao")
+! pos == 1
+
+pos = verify(string, "fo")
+! pos == 3
+
+pos = verify(string, "c++")
+! pos == 1
+
+pos = verify(string, "c++", back=.true.)
+! pos == 7
+
+pos = verify(string, string)
+! pos == 0
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Lgt function (lexical greater)
+
+Lexically compare the order of two character sequences being greater.
+
+#### Description
+
+The left-hand side, the right-hand side or both character sequences can
+be represented by a string type.
+This defines three procedures overloading the intrinsic `lgt` procedure.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):lgt(interface)]] (lhs, rhs)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+- `lhs`: Either scalar character value or string type. This argument is `intent(in)`.
+- `rhs`: Either scalar character value or string type. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a default logical scalar value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+logical :: res
+
+string = "bcd"
+res = lgt(string, "abc")
+! res .eqv. .true.
+
+res = lgt(string, "bcd")
+! res .eqv. .false.
+
+res = lgt(string, "cde")
+! res .eqv. .false.
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Llt function (lexical less)
+
+Lexically compare the order of two character sequences being less.
+
+#### Description
+
+The left-hand side, the right-hand side or both character sequences can
+be represented by a string type.
+This defines three procedures overloading the intrinsic `llt` procedure.
```

Comment:
```suggestion
This defines three procedures overloading the intrinsic `llt` procedure.
```
Is this sentence needed for a user?

Reply:
Yes, it's stating the obvious, but I wouldn't consider it too harmful.